src/cogent3/core/new_alignment.py

```diff
@@ -2374,3 +2379,559 @@ def _(
+            seqs[name], gaps[name] = seq, map
+
+        return cls(
+            seqs=seqs,
+            gaps=gaps,
+            moltype=moltype,
+            align_len=align_len,
+            check=False,
+        )
+
+    def __len__(self):
+        return self.align_len
+
+    @singledispatchmethod
+    def __getitem__(self, index: Union[str, int]) -> Aligned:
+        raise NotImplementedError(f"__getitem__ not implemented for {type(index)}")
+
+    @__getitem__.register
+    def _(self, index: str) -> Aligned:
+        adv = self.get_aligned_view(seqid=index)
+        return Aligned(data=adv, moltype=self.moltype)
+
+    @__getitem__.register
+    def _(self, index: int) -> Aligned:
+        return self[self.names[index]]
+
+    @property
+    def names(self) -> tuple[str]:
+        return self._names
+
+    @property
+    def alphabet(self) -> new_alphabet.CharAlphabet:
+        return self._alphabet
+
+    def get_aligned_view(self, seqid: str) -> AlignedDataView:
+        # refactor: design
+        # need to revisit what the variable is called i.e. parent_length
+        return AlignedDataView(seq=self, seqid=seqid, seq_len=self.align_len)
+
+    def get_gaps(self, seqid: str) -> numpy.ndarray:
+        return self.gaps[seqid]
+
+    def get_seq_array(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> numpy.ndarray:
+        """Return sequence corresponding to seqid as an array of indices.
+        start/stop are in alignment coordinates. Excludes gaps.
+        """
+        seq = self.seqs[seqid]
+        gaps = self.gaps[seqid]
+
+        # if there's no gaps, just slice the sequence
+        if len(seq) == self.align_len:
+            return seq[start:stop]
+
+        new_map = gaps[start:stop] if start is not None or stop is not None else gaps
+        seq_start = gaps.get_seq_index(start or 0)
+        seq_end = gaps.get_seq_index(stop or self.align_len)
+        return seq[seq_start:seq_end] if new_map.useful else seq[:0]
+
+    def get_gapped_seq_array(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> numpy.ndarray:
+        """Return sequence corresponding to seqid as an array of indices.
+        start/stop are in alignment coordinates. Includes gaps.
+        """
+        seq = self.seqs[seqid]
+        gaps = self.gaps[seqid]
+
+        # if there's no gaps, just slice the sequence
+        if len(seq) == self.align_len:
+            return seq[start:stop]
+
+        return self.moltype.most_degen_alphabet().gapped_by_map(seq, gaps)[start:stop]
+
+    def get_seq_str(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> str:
+        """Return sequence corresponding to seqid as a string.
+        start/stop are in alignment coordinates. Excludes gaps."""
+
+        return self.alphabet.from_indices(
```

Comment:
How is this method different to `get _gapped_seq_str`? It seems to me that those coordinates are alignment coordinates but the returned object is the underlying ungapped sequence. In that case you need the `IndelMap` object for conversion of alignment coordinates into sequence coordinates.

Reply:
@GavinHuttley `get_gapped_seq_str` includes gaps and `get_seq_str` does not. `get_seq_str` uses the `IndelMap` for coordinate conversion - (see `AlignedData.get_seq_array`)